File: Leetcode/238.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<int> productExceptSelf(vector<int>& nums) {

    vector<int> result;

    if  (count(nums.begin(), nums.end(), 0) >= 2){
        for (auto i: nums){
            result.push_back(0);
        }
    }else if (count(nums.begin(), nums.end(), 0) == 1){
        
        int val = 1;

        for (auto i : nums){
            if (i != 0){
                val *= i;
            }
        }

        for (auto i : nums){
            if (i != 0){
                result.push_back(0);
            }else{
                result.push_back(val);
            }
        }
    }else{
        int val = 1;

        for (auto i : nums){
            val *= i;
        }

        for (auto i : nums){
            result.push_back(val/i);
        }
    }
    return result;
    
}

int main(){
    vector<int> nums {1, 2, 3, 4};

    vector<int> result = productExceptSelf(nums);

    for (auto i: result){
        cout << i << endl;
    }
}

File: Leetcode/Python/238.py
def productExceptSelf(nums: list[int]) -> list[int]:
    result = []
    
    if nums.count(0) >= 2:
        result = [0] * len(nums)
    
    elif nums.count(0) == 1:
        # get mul except 0
        val = 1
        for i in nums:
            if i != 0:
                val *= i

        for i in nums:
            if i!= 0:
                result.append(0)
            else:
                result.append(val)
        
    else:
        val = 1
        for i in nums:
            val *= i
        
        for i in nums:
            result.append(int(val / i))
    
    return result

if __name__ == "__main__":
    nums = [1, 2, 3, 4]
    print(productExceptSelf(nums))